e-commerce-website/src/pages/ProductDetail/index.jsx

```diff
@@ -55,6 +58,39 @@ function ProductDetail(props) {
     })()
   }, [productParamId, dispatch])
 
+  useEffect(() => {
+    let newViewedProduct = null
+    const viewedProducts =
+      JSON.parse(localStorage.getItem(LocalStorage.viewedProducts)) || []
+
+    if (Object.keys(productDetail).length) {
+      newViewedProduct = {
+        id: productDetail.id,
+        image: productDetail.image,
+        name: productDetail.name,
+        price_before_discount: productDetail.price_before_discount,
+        price: productDetail.price,
+        rating: productDetail.rating,
+        sold: productDetail.sold,
+        place: {
+          name: productDetail.place.name
+        }
+      }
```

Comment:
```suggestion
      newViewedProduct = { ...productDetail }
```
Còn nhiều cách để copy 1 obj nữa, thay vì đi set từng giá trị như này

Reply:
dạ trong productDetail chứa nhiều thông tin không cần thiết cho viewed product nên em chỉ lấy 1 số trường cần dùng cho render product item thôi. Tại vì viewed product em lưu vào localstorage nên em nghĩ cần lượt bớt thông tin không cần thiết (ví dụ mô tả sản phẩm rất dài) để tiết kiệm localstorage